### ISSUE TITLE
CA1010: False positive for covariant generic interface

### ISSUE THREAD

**Issue by znakeeye** [dotnet/roslyn-analyzers#7710]
The following interface is perfectly valid:
```cs
public interface IMyCollection<out T> : ICollection, IEnumerable<T>, IEnumerable
{
}
```
However, the code analyzer produces `CA1010` since we did not implement `ICollection<T>`.

* Due to covariance, we **cannot** implement `ICollection<T>`
* If we instead implement `IReadOnlyCollection<T>` we get a conflict between `ICollection.Count` and `IReadOnlyCollection<T>.Count`.

![Image](https://github.com/user-attachments/assets/0ec0d315-4bf2-4c17-9a32-cf95d05730b0)

### Suggested fix
`CA1010` should not be produced when `ICollection` is implemented **and** the declared interface is covariant.
